lib/datadog/tracing/span_event.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require 'time'
+
+module Datadog
+  module Tracing
+    # SpanEvent represents an annotation on a span.
+    # @public_api
+    class SpanEvent
+      # @!attribute [r] name
+      #   @return [Integer]
+      attr_reader :name
+
+      # @!attribute [r] attributes
+      #   @return [Hash{String => String, Numeric, Boolean, Array<String, Numeric, Boolean>}]
+      attr_reader :attributes
+
+      # @!attribute [r] time_unix_nano
+      #   @return [Integer]
+      attr_reader :time_unix_nano
+
+      def initialize(
+        name,
+        attributes: nil,
+        time_unix_nano: nil
+      )
+        @name = name
+        @attributes = attributes || {}
+        @time_unix_nano = time_unix_nano || Core::Utils::Time.now.to_f * 1e9
```

Comment:
This writes a floating-point time into `@time_unix_nano`; I assume the time received from otel will be an integer. The expected type for the field should be documented somewhere, and given that Ruby's times do not exceed nanosecond precision, the current time in nanoseconds can be exactly obtained by
```
(Time.now.to_r * 1_000_000_000).to_i
```

Reply:
Nice catch. The current approach has some percission errors. This should improve it